src/fparser/two/tests/test_c99preprocessor.py

```diff
@@ -0,0 +1,569 @@
+    ref = '''
+SUBROUTINE FOO(BAR2)
+  INTEGER, INTENT(INOUT) :: BAR2
+  #ifdef MACRO_NAME
+  CALL FOOBAR2(BAR)
+  #else
+  CALL ABORT("error message")
+  #endif
+  BAR2 = BAR2 + 1
+END SUBROUTINE FOO
+'''.strip()
+    code = '''
+SUBROUTINE FOO(BAR2)
+  INTEGER, INTENT(INOUT) :: BAR2
+#ifdef MACRO_NAME
+  CALL FOOBAR2(BAR)
+#else
+  CALL ABORT("error message")
+#endif
+  BAR2 = BAR2 + 1
+END SUBROUTINE FOO
+'''
+    reader = get_reader(code)
+    result = f2003_parser(reader)
+    assert str(result) == ref
+
+
+def test_parse_ifdef_in_subroutine():
+    f2003_parser = ParserFactory().create(std='f2003')
+    ref = '''
+SUBROUTINE FOO(BAR3)
+  #ifdef MACRO_NAME
+  INTEGER, INTENT(INOUT) :: BAR3
+  CALL FOOBAR3(BAR)
+  #endif
+END SUBROUTINE FOO
+'''.strip()
+    code = '''
+SUBROUTINE FOO(BAR3)
+#ifdef MACRO_NAME
+  INTEGER, INTENT(INOUT) :: BAR3
+  CALL FOOBAR3(BAR)
+#endif
+END SUBROUTINE FOO
+'''
+    reader = get_reader(code)
+    result = f2003_parser(reader)
+    assert str(result) == ref
+
+
+def test_parse_ifdef_elif_in_subroutine():
+    f2003_parser = ParserFactory().create(std='f2003')
+    ref = '''
+#define MACRO
+SUBROUTINE BAR
+  #if defined(MACRO)
+  CALL sub1
+  #elif FOO
+  CALL sub2
+  #else
+  CALL sub3
+  #endif
+END SUBROUTINE BAR'''.strip()
+    code = '''
+#define MACRO
+SUBROUTINE BAR
+#if defined(MACRO)
+  call sub1()
+#elif FOO
+  call sub2()
+#else
+  call sub3()
+#endif
+END SUBROUTINE BAR
+'''
+    reader = get_reader(code)
+    result = f2003_parser(reader)
+    assert str(result) == ref
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', ['#include "filename.inc"',
+                                  '   #   include  "filename.inc"  '])
+def test_include_stmt_quotes(line):
+    '''Test that #include is recognized'''
+    ref = '#include "filename.inc"'
+    result = Cpp_Include_Stmt(line)
+    assert str(result) == ref
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', ['#include <filename.inc>',
+                                  '   #   include  <filename.inc>  '])
+def test_include_stmt_bracket(line):
+    '''Test that #include is recognized'''
+    ref = '#include "filename.inc"'
+    result = Cpp_Include_Stmt(line)
+    assert str(result) == ref
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line', [
+    None, '', '  ', '#includ', '#includ "x"', '#include', '#include ""',
+    "#include 'x'", '#include "x', '#include x"', '#include x',
+    '#include x"x"', '#include "x"x', 'x #include "x"', '#includex "x"',
+    "#include 'abc'", '#include " a.inc"', '#include "      "'])
+def test_incorrect_include_stmt(line):
+    '''Test that incorrectly formed #include statements raise exception'''
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = Cpp_Include_Stmt(line)
+    assert "Cpp_Include_Stmt: '{0}'".format(line) in str(excinfo.value)
+
+
+@pytest.mark.usefixtures("f2003_create")
+@pytest.mark.parametrize('line, ref', [
```

Comment:
Almost all of the inputs and outputs are the same with parametrize. You could either split the tests into identical ones and ones with extra spaces and then for the identical just pass in the reference, or you could just pass in the original for all of them and then check with e.g. `' '.join(line.split())`. Either way there is no need for both `line` and `ref` for most (or all) of the tests.